naga/src/front/wgsl/parse/mod.rs

```diff
@@ -1619,22 +1619,41 @@ impl Parser {
         lexer: &mut Lexer<'a>,
         ctx: &mut ExpressionContext<'a, '_, '_>,
         block: &mut ast::Block<'a>,
+        brace_nesting_level: u8,
     ) -> Result<(), Error<'a>> {
         self.push_rule_span(Rule::Statement, lexer);
         match lexer.peek() {
             (Token::Separator(';'), _) => {
                 let _ = lexer.next();
                 self.pop_rule_span(lexer);
-                return Ok(());
             }
-            (Token::Paren('{'), _) => {
-                let (inner, span) = self.block(lexer, ctx)?;
+            (Token::Paren('{'), brace_span) => {
+                // From [spec.](https://gpuweb.github.io/gpuweb/wgsl/#limits):
+                //
+                // > § 2.4. Limits
+                // >
+                // > …
+                // >
+                // > Maximum nesting depth of brace-enclosed statements in a function[:] 127
+                //
+                // _However_, we choose 64 instead because (a) it avoids stack overflows in CI and
+                // (b) we expect the limit to be decreased to 63 based on this conversation in
+                // WebGPU CTS upstream:
+                // <https://github.com/gpuweb/cts/pull/3389#discussion_r1543742701>
```

Comment:
FTR: https://github.com/gpuweb/cts/pull/3389#discussion_r1543742701

Reply:
Motivation is @jimblandy's comment at <https://github.com/gfx-rs/wgpu/issues/4352#issuecomment-2025545672>.